src/prog_models/utils/containers.py

```diff
@@ -29,14 +30,24 @@ def __init__(self, keys: list, data: Union[dict, np.array]):
                 data = data[np.newaxis].T
             self.matrix = data
         elif isinstance(data, (dict, DictLikeMatrixWrapper)):
-          # ravel is used to prevent vectorized case, where data[key] returns multiple values,  from resulting in a 3D matrix
-          self.matrix = np.array(
-             [
-                  np.ravel([data[key]]) if key in data else [None] for key in keys
-              ], dtype=np.float64)
+            # ravel is used to prevent vectorized case, where data[key] returns multiple values,  from resulting in a 3D matrix
+            self.matrix = np.array(
+                [
+                    np.ravel([data[key]]) if key in data else [None] for key in keys
+                ], dtype=np.float64)
         else:
             raise ProgModelTypeError(f"Data must be a dictionary or numpy array, not {type(data)}")
 
+    @property
+    def frame(self) -> pd.DataFrame:
+        """
+            frame -- Getter for pd.DataFrame
+
+            Returns: pd.DataFrame
+        """
+        self._frame = pd.DataFrame(self.matrix.T, columns=self._keys)
```

Comment:
What is the purpose of _frame as a property instead of a local variable? _frame is recalculated every time you access frame. Are you planning on using caching in a future version.